site/docs/transform/impute.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: docs
+title: Impute
+permalink: /docs/impute.html
+---
+
+To impute missing data in Vega-Lite, you can either use the `impute` transform either inside an [encoding field definition](#encoding) or inside an [`transform`](#transform) array.
+
+Impute transform in Vega-Lite groups data and determines missing values of the `key` field within each group.  For each missing value in each group, a new tuple will be generated
+with the `impute`d field generated based on specified imputation method (by using a constant `value` or by calculating statistics such as mean within each group).
+
+## Documentation Overview
+{:.no_toc}
+
+- TOC
+{:toc}
+
+{:#encoding}
+## Impute in Encoding Field Definition
+
+{: .suppress-error}
+```json
+// A Single View Specification
+{
+  "data": ... ,
+  "mark": ... ,
+  "encoding": {
+    "x": {
+      "field": ...,
+      "type": "quantitative",
+      "impute": ...,               // Impute
+      ...
+    },
+    "y": ...,
+    ...
+  },
+  ...
+}
+```
+
+The `impute` property of [a field definition](encoding.html#field-def) can be used to generate new data objects in place of the missing data.
+
+{% include table.html props="value,method,frame" source="ImputeParams" %}
+
+For `impute` in encoding, grouping fields and the key field (for identifying missing values) are automatically determined. Values are automatically grouped by the specified fields of [mark property channels](encoding.html#mark-prop), [key channel](encoding.html#key) and [detail channel](encoding.html#detail). If x-field is `impute`d, y-field is the key field. Basically, any missing y-value in each group will lead to a new tuple imputed, and vice versa.
+
+In this example, the data object where `"x"` is `3` and `"c"` is `1` is missing. So we can impute the missing data and give the imputed field a value of `0`.
```

Comment:
This is the first example in the page.  Better explain here c is the groupby field and then x is the key field.  Then say that for group c=1, x=3 is missing.